src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -89,5 +112,62 @@ internal async Task<AuthenticationResult> ExecuteTestAsync(CancellationToken can
         {
             return await ExecuteAsync(cancellationToken).ConfigureAwait(false);
         }
+
+        private async Task UpdateRequestWithAccountAsync()
+        {
+            IAccount account = await GetAccountFromParamsOrLoginHintAsync(
+                _silentParameters.Account,
+                _silentParameters.LoginHint).ConfigureAwait(false);
+
+            AuthenticationRequestParameters.Account = account;
+
+            AuthenticationRequestParameters.Authority = Authority.CreateAuthorityForRequest(
+                ServiceBundle.Config.AuthorityInfo,
+                AuthenticationRequestParameters.AuthorityOverride,
+                account?.HomeAccountId?.TenantId);
+        }
+
+        private async Task<IAccount> GetSingleAccountForLoginHintAsync(string loginHint)
+        {
+            if (!string.IsNullOrEmpty(loginHint))
+            {
+                IReadOnlyList<IAccount> accounts = (await CacheManager.GetAccountsAsync().ConfigureAwait(false))
+                    .Where(a => !string.IsNullOrWhiteSpace(a.Username) &&
+                           a.Username.Equals(loginHint, StringComparison.OrdinalIgnoreCase))
+                    .ToList();
+
+                if (accounts.Count() == 0)
```

Comment:
>accounts [](start = 20, length = 8)

could accounts be null?

Reply:
No, Linq always returns empty collection instead of null (and so should we). But you did spot a problem actually, I should use `.Count` instead of `.Count()` since `.Count()` is more expensive.